app/server/auth/signup/route.ts

```diff
@@ -9,14 +9,15 @@ connect()
 export async function POST(req: NextRequest) {
     try {
         const reqBody = await req.json();
-        
-        const {userName, password, email} = reqBody;
 
-        const userData = await User.findOne({email})
-        if(userData){
-            return NextResponse.json({error: 'User already exists'}, {status: 400})
+        const { userName, password, email } = reqBody;
+
+        const userData = await User.findOne({ $or: [email, userName] })
```

Comment:
return right message ->

- if userName exist the return Username already exist use different username.
- if email exist the return User already exist with this email.